File: frontend/src/app/shared/models/vehicle/vehicle.dto.ts
import { VehicleCategory } from '../../enums/vehicle/vehicle-category.enum';
import { VehicleMotor } from '../../enums/vehicle/vehicle-motor.enum';
import { VehicleStatus } from '../../enums/vehicle/vehicle-status.enum';

export interface VehicleDTO {
  id?: number;
  registration: string;
  brand?: string;
  model: string;
  seats: number;
  isCompany?: boolean;
  co2PerKm?: number;
  vehicleStatus?: VehicleStatus;
  vehicleCategory?: VehicleCategory;
  vehicleMotor?: VehicleMotor;
  urlImage?: string;
}

File: src/main/java/com/diginamic/groupe1/transport/dto/VehicleDTO.java
package com.diginamic.groupe1.transport.dto;

import com.diginamic.groupe1.transport.enums.VehicleCategory;
import com.diginamic.groupe1.transport.enums.VehicleMotor;
import com.diginamic.groupe1.transport.enums.VehicleStatus;
import com.diginamic.groupe1.transport.validation.ValidationGroups;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class VehicleDTO {

    private Long id;

    @NotNull(message = "registration error")
    private String registration;

    private String brand;

    @NotNull(message = "model error")
    private String model;

    @NotNull(message = "seats error")
    private Integer seats;

    private Boolean isCompany;

    private Integer co2PerKm;

    private VehicleStatus vehicleStatus;

    private VehicleCategory vehicleCategory;

    private VehicleMotor vehicleMotor;
}
